common/txmgr/types/config.go

```diff
@@ -0,0 +1,58 @@
+package types
+
+import "time"
+
+type TxmConfig[UNIT any] interface {
+	BroadcasterConfig[UNIT]
+	ConfirmerConfig[UNIT]
+	ResenderConfig
+	ReaperConfig
+
+	SequenceAutoSync() bool
+	UseForwarders() bool
+	MaxQueuedTransactions() uint64
+}
+
+type BroadcasterConfig[UNIT any] interface {
```

Comment:
Any reason we want to have separate config for Broadcaster, Confirmer, Resender and Reaper?
I mean, why not club everything under a single TxmConfig.

The advantage then we get, is that when chain specific code has to instantiate all components, it can just pass in TxmConfig. They don't have to understand all details of different subcomponents of Txm.
Just makes the Txm public interface look more simpler.

Anyways, I don't have strong opinion on this. We can leave it as it is too for now, consider later, if needed.

Reply:
One of the [Go Proverbs](https://go-proverbs.github.io/) is:

> The bigger the interface, the weaker the abstraction.

We can always trivially include additional interfaces or combinations of interfaces in order to guide an implementation, but by using smaller interfaces that only include what is actually used for each of these case, we have more flexibility and less noise, conflicts, unused methods/params, etc.

Reply:
yeah, from past experience I've learned more narrowly scoped interfaces are usually cleaner and easier to understand rather than passing a big blob of interfaces to each component even if they don't need anything

Reply:
Ok, thats fine by me